packages/core/src/jsonld/documentLoader.ts

```diff
@@ -35,14 +36,24 @@ export class DocumentLoader {
         };
       }
 
-      if (url.startsWith('did:tangle:')) {
-        const document = await this.registry.fetch(url);
+      if (isDidUrl(url)) {
+        const decoded = decodeFromDidUrl(url);
+        if (decoded != null && decoded.method === 'tangle') {
+          let document = (await this.registry.fetch(decoded.did)) as object;
 
-        return {
-          document,
-          documentUrl: url,
-          contextUrl: null,
-        };
+          if (decoded.fragment) {
```

Comment:
What will return get if this `if` check fails?

Reply:
If the document loader get this DID URL `did:tangle:123456789abcdefghi`.

And fetch the following `document` from `registry`.
```
{
  "@context": "https://w3id.org/did/v1",
  "id": "did:tangle:123456789abcdefghi",
   "publicKey": [{
    "id": "did:tangle:123456789abcdefghi#keys-1",
    "type": "RsaVerificationKey2018",
    "controller": "did:example:123456789abcdefghi",
    "publicKeyPem": "-----BEGIN PUBLIC KEY...END PUBLIC KEY-----\r\n"
  }]
}
```
Because the given DID URL doesn't contain the fragment, it will return the above `document`.